apps/meteor/tests/unit/server/publications/messages.spec.ts

```diff
@@ -0,0 +1,282 @@
+	'meteor/check': {
+		check: sinon.stub(),
+	},
+	'meteor/meteor': {
+		'Meteor': {
+			methods: sinon.stub(),
+			Error: mockMeteorError,
+		},
+		'@global': true,
+	},
+	'@rocket.chat/models': {
+		Messages: messagesMock,
+	},
+});
+
+describe('extractTimestampFromCursor', () => {
+	it('should return the correct timestamp', () => {
+		const cursor = new Date('2024-01-01T00:00:00.000Z').getTime().toString();
+		const timestamp = extractTimestampFromCursor(cursor);
+		expect(timestamp).to.be.an.instanceOf(Date);
+		expect(timestamp.getTime()).to.equal(parseInt(cursor, 10));
+	});
+});
+
+describe('mountCursorQuery', () => {
+	const mockDate = new Date('2024-01-01T00:00:00.000Z').getTime();
+
+	it('should return a query with $gt when next is provided', () => {
+		const result = mountCursorQuery({ next: mockDate.toString() });
+		expect(result.query).to.deep.equal({ $gt: new Date(mockDate) });
+	});
+
+	it('should return a query with $lt when previous is provided', () => {
+		const result = mountCursorQuery({ previous: mockDate.toString() });
+		expect(result.query).to.deep.equal({ $lt: new Date(mockDate) });
+	});
+
+	it('should return a query with $gt and sort when neither next nor previous is provided', () => {
+		const result = mountCursorQuery({ count: 10 });
+		expect(result.query).to.deep.equal({ $gt: new Date(0) });
+		expect(result.options).to.deep.equal({ sort: { ts: -1 } });
+	});
+});
+
+describe('mountCursorFromMessage', () => {
+	it('should return the updated timestamp for UPDATED type', () => {
+		const message: Pick<IMessage, '_updatedAt'> = {
+			_updatedAt: new Date('2024-01-01T00:00:00Z'),
+		};
+
+		const result = mountCursorFromMessage(message, 'UPDATED');
+		expect(result).to.equal(`${message._updatedAt.getTime()}`);
+	});
+
+	it('should return the deleted timestamp for DELETED type', () => {
+		const message: Partial<IMessage> & { _deletedAt: Date } = {
+			_deletedAt: new Date('2024-01-01T00:00:00Z'),
+		};
+
+		const result = mountCursorFromMessage(message, 'DELETED');
+		expect(result).to.equal(`${message._deletedAt.getTime()}`);
+	});
+
+	it('should throw an error if DELETED type and _deletedAt is not present', () => {
+		const message: Pick<IMessage, '_updatedAt'> = {
+			_updatedAt: new Date('2024-01-01T00:00:00Z'),
+		};
+
+		expect(() => mountCursorFromMessage(message, 'DELETED')).to.throw(mockMeteorError, 'Cursor not found');
+	});
+});
+
+describe('mountNextCursor', () => {
+	it('should return a cursor for the most recent message when messages are present', () => {
+		// Messages are already sorted by descending order
+		const messages = [{ _updatedAt: new Date('2024-10-01T10:00:00Z') }, { _updatedAt: new Date('2024-10-01T09:00:00Z') }];
+		const type = 'UPDATED';
+		const result = mountNextCursor(messages, type);
+		expect(result).to.equal(`${messages[0]._updatedAt.getTime()}`);
+	});
+
+	it('should reverse messages if next is provided', () => {
```

Comment:
We could improve this title, it seems to me the array of messages will be reversed...